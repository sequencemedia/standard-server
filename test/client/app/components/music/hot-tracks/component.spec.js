import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import Component from '@sequencemedia/app/components/music/hot-tracks/component'

Enzyme.configure({ adapter: new Adapter() })

jest.mock('@sequencemedia/app/components/common/playlists/order', () => {
  const MockOrder = () => (
    <div className='mock-order'>
      Mock Order
    </div>
  )

  return {
    __esModule: true,
    default: MockOrder
  }
})
jest.mock('@sequencemedia/app/components/common/playlists/reset', () => {
  const MockReset = () => (
    <div className='mock-reset'>
      Mock Reset
    </div>
  )

  return {
    __esModule: true,
    default: MockReset
  }
})
jest.mock('@sequencemedia/app/components/common/playlists', () => {
  const MockPlaylists = () => (
    <div className='mock-playlists'>
      Mock Latest
    </div>
  )

  return {
    __esModule: true,
    default: MockPlaylists
  }
})

const MOCK_ITEMS = [
  {
    kind: 'MOCK KIND',
    url: 'MOCK URL',
    name: 'MOCK NAME',
    artworkUrl100: 'MOCK ARTWORK URL 100'
  }
]

describe('shinkansen-cogs/cogs/checkbox', () => {
  describe('<Component />', () => {
    describe('With required props', () => {
      const component = (
        <Component
          feedType='MOCK FEED TYPE'
          onMount={jest.fn()}
          onMountOrderBy={jest.fn()}
          onClickLatest={jest.fn()}
          onClickLatestOrderBy={jest.fn()}
        />
      )

      it('renders', () => {
        return expect(toJSON(shallow(component)))
          .toMatchSnapshot()
      })
    })

    describe('With additional props', () => {
      const component = (
        <Component
          feedType='MOCK FEED TYPE'
          order='MOCK ORDER'
          items={MOCK_ITEMS}
          onMount={jest.fn()}
          onMountOrderBy={jest.fn()}
          onClickLatest={jest.fn()}
          onClickLatestOrderBy={jest.fn()}
        />
      )

      it('renders', () => {
        return expect(toJSON(shallow(component)))
          .toMatchSnapshot()
      })
    })
  })
})
