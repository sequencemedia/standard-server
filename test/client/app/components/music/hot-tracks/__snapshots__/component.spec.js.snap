// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shinkansen-cogs/cogs/checkbox <Component /> With additional props renders 1`] = `
<div
  className="MOCK FEED TYPE"
>
  <h3>
    Hot Tracks
  </h3>
  <MockOrder
    feedType="MOCK FEED TYPE"
    order="MOCK ORDER"
  />
  <MockReset
    feedType="MOCK FEED TYPE"
    order="MOCK ORDER"
  />
  <MockPlaylists
    items={
      Array [
        Object {
          "artworkUrl100": "MOCK ARTWORK URL 100",
          "kind": "MOCK KIND",
          "name": "MOCK NAME",
          "url": "MOCK URL",
        },
      ]
    }
  />
</div>
`;

exports[`shinkansen-cogs/cogs/checkbox <Component /> With required props renders 1`] = `
<div
  className="MOCK FEED TYPE"
>
  <h3>
    Hot Tracks
  </h3>
  <MockOrder
    feedType="MOCK FEED TYPE"
    order="by-none"
  />
  <MockReset
    feedType="MOCK FEED TYPE"
    order="by-none"
  />
  <MockPlaylists
    items={Array []}
  />
</div>
`;
